src/Form/Type/VichImageType.php

```diff
@@ -71,7 +71,7 @@ public function buildView(FormView $view, FormInterface $form, array $options):
         $view->vars['image_uri'] = null;
         $view->vars['download_uri'] = null;
 
-        if (null !== $object) {
+        if ($options['download_uri'] && $object) {
```

Comment:
Is necessary to prevent generating download URL for a new object, which it's not stored in db and has not data.

Reply:
`fileNameProperty: 'sourceFile.filePath',`

sourceFile can be null, because it's not required.

PropertyAccessor requires a graph of objects or arrays to operate on, but it found type "NULL" while trying to traverse path "sourceFile.filePath" at property "filePath".

`
in vendor/symfony/property-access/PropertyAccessor.php (line 325)
in vendor/symfony/property-access/PropertyAccessor.php -> readPropertiesUntil (line 104)
in vendor/vich/uploader-bundle/src/Mapping/PropertyMapping.php -> getValue (line 148)
in vendor/vich/uploader-bundle/src/Mapping/PropertyMapping.php -> readProperty (line 94)
in vendor/vich/uploader-bundle/src/Storage/AbstractStorage.php -> getFileName (line 129)
in vendor/vich/uploader-bundle/src/Storage/AbstractStorage.php -> getFilename (line 88)
in vendor/vich/uploader-bundle/src/Form/Type/VichFileType.php -> resolveUri (line 143)
in vendor/vich/uploader-bundle/src/Form/Type/VichFileType.php -> resolveUriOption (line 120)
in vendor/symfony/form/ResolvedFormType.php -> buildView (line 116) 
`

Reply:
Can't you just pass `false` as `download_uri` then?

Reply:
I do it, but VichFileType still tries to build URL.

Reply:
can you debug the point where it tries to do so?

Reply:
I pasted trace above. While form is rendering the exception is thrown.

If you have entity A with `#[ORM\ManyToOne(cascade: ['persist'])]` to entity B and the form is created for entity A, `$form->getParent()->getData()` returns entity A and with `fileNameProperty: 'b.filePath'` `PropertyAccessor` tries to get B, but `$a->getB()` can be null. In the current code there is only one condition `if (null !== $object)`.  A is not null, but B can be null. To prevent accessing B by `PropertyAccessor`  we need to disable generating URL by `download_uri = false`. It works in `VichFileType` but doesn't in `VichImageType`.

Reply:
The condition seems too early to me.
It should be before this line:
`$view->vars['download_uri'] = $this->resolveUriOption($options['download_uri'], $object, $form);`

Reply:
I'm going to resolve my problem by dynamic initialization of B entity, so we can close this PR.